lib/src/read-contents/read-buffer.js

```diff
@@ -1,23 +1,41 @@
 'use strict';
 
 var fs = require('graceful-fs');
+var iconv = require('iconv-lite');
 var removeBomBuffer = require('remove-bom-buffer');
 
+var getCodec = require('../../codecs');
+
 function bufferFile(file, optResolver, onRead) {
+  var encoding = optResolver.resolve('encoding', file);
+  var codec = getCodec(encoding);
+  if (encoding && !codec) {
+    return onRead(new Error('Unsupported encoding: ' + encoding));
+  }
+
   fs.readFile(file.path, onReadFile);
 
-  function onReadFile(readErr, data) {
+  function onReadFile(readErr, contents) {
     if (readErr) {
       return onRead(readErr);
     }
 
-    var removeBOM = optResolver.resolve('removeBOM', file);
-    if (removeBOM) {
-      file.contents = removeBomBuffer(data);
-    } else {
-      file.contents = data;
+    if (encoding) {
+      var removeBOM = codec.bomAware && optResolver.resolve('removeBOM', file);
+
+      if (codec.enc !== 'utf8') {
+        contents = iconv.decode(contents, encoding, { stripBOM: false });
+        removeBOM = removeBOM && contents[0] === '\ufeff';
+        contents = iconv.encode(contents, 'utf8');
```

Comment:
Inverse of my `write` question about translating to utf8.  Also, can node's `fs.readFile` read any encoding and store it in a buffer?  I thought they only supported a few encodings.  As you probably can tell, I don't understand this stuff very well.